nautobot/core/views/mixins.py

```diff
@@ -0,0 +1,804 @@
+        If not, return instantiated form using form_class.
+        """
+        form = getattr(self, f"{self.action}_form", None)
+        if not form:
+            form_class = self.get_form_class()
+            if not form_class:
+                self.logger.debug(f"{self.action}_form_class is not defined")
+                return None
+            form = form_class(*args, **kwargs)
+        return form
+
+    def get_form_class(self, **kwargs):
+        """
+        Helper function to get form_class for different views.
+        """
+        if self.action in ["create", "update"]:
+            form_class = getattr(self, "form_class", None)
+        elif self.action == "bulk_create":
+
+            class BulkCreateForm(BootstrapMixin, Form):
+                csv_data = CSVDataField(
+                    from_form=self.bulk_create_form_class, widget=Textarea(attrs=self.bulk_create_widget_attrs)
+                )
+                csv_file = CSVFileField(from_form=self.bulk_create_form_class)
+
+            form_class = BulkCreateForm
+        else:
+            form_class = getattr(self, f"{self.action}_form_class", None)
+
+        if not form_class:
+            if self.action == "bulk_destroy":
+
+                class BulkDestroyForm(ConfirmationForm):
+                    pk = ModelMultipleChoiceField(queryset=self.queryset, widget=MultipleHiddenInput)
+
+                return BulkDestroyForm
+            else:
+                # Check for request first and then kwargs for form_class specified.
+                form_class = self.request.data.get("form_class", None)
+                if not form_class:
+                    form_class = kwargs.get("form_class", None)
+        return form_class
+
+    def form_save(self, form, **kwargs):
+        """
+        Generic method to save the object from form.
+        Should be overriden by user if customization is needed.
+        """
+        return form.save()
+
+
+class ObjectDetailViewMixin(NautobotViewSetMixin, mixins.RetrieveModelMixin):
+    def get_changelog_url(self, instance):
+        """Return the changelog URL for a given instance."""
+        meta = self.queryset.model._meta
+
+        # Don't try to generate a changelog_url for an ObjectChange.
+        if not issubclass(self.queryset.model, ChangeLoggedModel):
+            return None
+
+        route = f"{meta.app_label}:{meta.model_name}_changelog"
+        if meta.app_label in settings.PLUGINS:
+            route = f"plugins:{route}"
+
+        # Iterate the pk-like fields and try to get a URL, or return None.
+        fields = ["pk", "slug"]
+        for field in fields:
+            if not hasattr(instance, field):
+                continue
+
+            try:
+                return reverse(route, kwargs={field: getattr(instance, field)})
+            except NoReverseMatch:
+                continue
+
+        # This object likely doesn't have a changelog route defined.
+        return None
```

Comment:
This can go away, right? Since it's been moved to be called from the model instead? See: https://github.com/nautobot/nautobot/pull/1902/files#diff-4932b9507690a60fb07d0e2a2dbb48547dfe5c78dc55985ff4dfddb31c84a5bdR329-R353